parquet-hadoop/src/test/java/org/apache/parquet/format/converter/TestParquetMetadataConverter.java

```diff
@@ -580,6 +583,42 @@ private void testBinaryStats(StatsHelper helper) {
     Assert.assertTrue(roundTripStats.isEmpty());
   }
 
+  @Test
+  public void testBinaryStatsWithTruncation() {
+    int defaultTruncLen = ParquetProperties.DEFAULT_STATISTICS_TRUNCATE_LENGTH;
+    int[] validLengths = {1, 2, 16, 64, defaultTruncLen - 1, defaultTruncLen - 1};
+    for (int len : validLengths) {
+      testBinaryStatsWithTruncation(len);
+    }
+
+    int[] invalidLengths = {-1, 0, defaultTruncLen + 1};
```

Comment:
`defaultTruncLen + 1` is invalid only because `ParquetProperties.DEFAULT_STATISTICS_TRUNCATE_LENGTH` is `Integer.MAX_VALUE` so it overflows. Meanwhile, semantically a larger value of the default one shall be supported.

Reply:
Make sense. I will make it as "-1, 0, Integer.MAX_VALUE + 1"

Reply:
Still not fixed.

Reply:
Don't know how it is slipped. Fix it now.